src/main/groovy/io/openliberty/tools/gradle/tasks/GenerateFeaturesTask.groovy

```diff
@@ -0,0 +1,348 @@
+                }
+                configDocument.writeXMLDocument(newServerXmlSrc);
+                logger.debug("Created file " + newServerXmlSrc);
+                // Add a reference to this new file in existing server.xml.
+                def serverXml = findConfigFile("server.xml", server.serverXmlFile);
+                def doc = getServerXmlDocFromConfig(serverXml);
+                logger.debug("Xml document we'll try to update after generate features doc=" + doc + " file=" + serverXml);
+                addGenerationCommentToConfig(doc, serverXml);
+
+                logger.lifecycle("Generated the following features: " + missingLibertyFeatures);
+                // use logger.lifecycle so that message appears without --info tag on
+            } else {
+                logger.lifecycle("No additional features were generated.");
+                if (newServerXmlSrc.exists()) {
+                    // generated-features.xml exists but no additional features were generated
+                    // create empty features list with comment
+                    ServerConfigXmlDocument configDocument = ServerConfigXmlDocument.newInstance();
+                    configDocument.createComment(HEADER);
+                    Element featureManagerElem = configDocument.createFeatureManager();
+                    configDocument.createComment(featureManagerElem, NO_NEW_FEATURES_COMMENT);
+                    configDocument.writeXMLDocument(newServerXmlSrc);
+                }
+            }
+        } catch (ParserConfigurationException | TransformerException | IOException e) {
+            logger.debug("Exception creating the server features file", e);
+            throw new GradleException("Automatic generation of features failed. Error attempting to create the " + GENERATED_FEATURES_FILE_NAME + ". Ensure your id has write permission to the server installation directory.", e);
+        }
+    }
+
+    /**
+     * Gets the binary scanner jar file from the local cache.
+     * Downloads it first from connected repositories such as Maven Central if a newer release is available than the cached version.
+     * Note: Maven updates artifacts daily by default based on the last updated timestamp. Users should use 'mvn -U' to force updates if needed.
+     *
+     * @return The File object of the binary scanner jar in the local cache.
+     * @throws PluginExecutionException
+     */
+    private File getBinaryScannerJarFromRepository() throws PluginExecutionException {
+        try {
+            return ArtifactDownloadUtil.downloadBuildArtifact(project, BINARY_SCANNER_MAVEN_GROUP_ID, BINARY_SCANNER_MAVEN_ARTIFACT_ID, BINARY_SCANNER_MAVEN_TYPE, BINARY_SCANNER_MAVEN_VERSION);
+        } catch (Exception e) {
+            throw new PluginExecutionException("Could not retrieve the binary scanner jar. Ensure you have a connection to Maven Central or another repository that contains the jar configured in your build.gradle: " + e.getMessage(), e);
```

Comment:
Consider whether this should just be a warning rather than a plugin execution exception. They could have all the necessary features in their server.xml already. Should not finding the binary scanner really cause a failure?